src/dsp_tools/fast_xmlupload/process_files.py

```diff
@@ -19,7 +19,10 @@
 
 from dsp_tools.models.exceptions import BaseError
 
+# default unconfigured logger created when the module is initialized (e.g. for testing, or when the module is imported)
+# if this module is used as part of the dsp-tools CLI, this default logger will be replaced by the one configured in dsp_tools.py
 logger = logging.getLogger(__name__)
```

Comment:
ok, I see this now... but I'm still not sure I like it. To me it would seem more streamlined to have a logger instance per file, and ensuring the configuration by means of a simple log util

Reply:
Is that possible? My first approach was to make one method in `shared.py`, where a logger is created and configured to log to `~/.dsp-tools`. And then every module would call that method, when it is initialized. 

But I ran into the following problem: One single call to a DSP-TOOLS CLI will initialize **ALL** modules, so the log file is opened a dozen times, so every log call is logged a dozen times...

Reply:
[test-logging.zip](https://github.com/dasch-swiss/dsp-tools/files/11567456/test-logging.zip)

Here's a little test project I created. to me it seems to work - maybe you can try it out. Unless I'm misunderstanding, I think this would solve the problem you wanted to address, and still be to my taste. :)  
(but maybe I'm also misunderstanding)

One case in which I don't know what would happen, is if two instances of DSP-Tools are running at the same time? Would they log to the same file? Would that work without issues?  
But I suppose that's a question that applies to your solution too, right?